boto/iam/connection.py

```diff
@@ -1743,6 +1743,56 @@ def create_virtual_mfa_device(self, path, device_name):
         }
         return self.get_response('CreateVirtualMFADevice', params)
 
+    def delete_virtual_mfa_device(self, serial_number):
+        """
+        Deletes a virtual MFA device for the AWS account.
+        You must deactivate a user's virtual MFA device
+        before you can delete it.
+
+        :type serial_number: string
+        :param serial_number: The serial number which uniquely identifies
+            the MFA device. For virtual MFA devices, the serial number is
+            the same as the ARN.
+        """
+        params = {
+            'SerialNumber': serial_number
+        }
+        return self.get_response('DeleteVirtualMFADevice', params)
+
+    def get_all_virtual_mfa_devices(self, assignment_status=None, marker=None, max_items=None):
```

Comment:
Probably should set assignment_status='Any' as the default here if it is going to be the default for None anyway?

Reply:
As 'AssignmentStatus' request param is optional, we want to test:
- No 'AssignmentStatus' request param in the request to see if Server will return the list of any status of virtual MFAs.
- 'AssignmentStatus' request param with 'Assigned/Unassigned/Any/other words, e.g. hoge' in the request to see if Sever will behave the corresponding response.